### PR TITLE
[Java][Bazel]  Refine auto-generated pom files

### PR DESCRIPTION
**Pull request by raulchen**
## What do these changes do?

- Add comments.
- Consolidate code.
- Remove versions in super pom.

## Linter

- [x] I've run `scripts/format.sh` to lint the changes in this PR.

### CODE REVIEW

File: bazel/ray.bzl [hunk 1]
```diff
@@ -53,12 +53,13 @@ def define_java_module(name, additional_srcs = [], additional_resources = [], de
...
-
-def gen_java_pom_file(name):
     pom_file(
         name = "org_ray_ray_" + name + "_pom",
         targets = [
             ":org_ray_ray_" + name,
         ],
         template_file = name + "/pom_template.xml",
+        substitutions = {
+            "{auto_gen_header}": "<!-- This file is auto-generated by Bazel from pom_template.xml, do not modify it. -->",
```
Comment (guoyuhong) [inline bazel/ray.bzl]: It looks like that this header caused the test failure.

**Review by guoyuhong — COMMENTED** [review]
Thanks. This looks good. After this PR, there will be not pom file change after `./build.sh -l java`, right? Have you test this change with Maven?

File: bazel/ray.bzl (hunk 1 above)
Comment (raulchen) [reply bazel/ray.bzl]: the failure is because this comment cannot be the first line. fixed.

**Comment by raulchen** [comment]
yes, the changes in pom files are also committed. and maven will be tested in CI.

**Review by guoyuhong — APPROVED** [review]

File: java/runtime/pom.xml [hunk 2]
```diff
@@ -21,53 +22,60 @@
...
-    <dependency>
-      <groupId>com.github.davidmoten</groupId>
-      <artifactId>flatbuffers-java</artifactId>
-    </dependency>
-    <dependency>
-      <groupId>redis.clients</groupId>
-      <artifactId>jedis</artifactId>
-    </dependency>
     <dependency>
       <groupId>org.apache.arrow</groupId>
```
Comment (jovany-wang) [inline java/runtime/pom.xml]: These `dependency` items should have the same indentation as those you changed.

Comment (raulchen) [reply java/runtime/pom.xml]: The below ones are not indented, because the tool doesn't support that.

**Pull request merged**